tls/s2n_server_new_session_ticket.c

```diff
@@ -313,6 +336,16 @@ S2N_RESULT s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_st
     RESULT_GUARD(s2n_resume_encrypt_session_ticket(conn, output));
     RESULT_GUARD_POSIX(s2n_stuffer_write_vector_size(&ticket_size));
 
+    /* Come back to the ticket lifetime field and write the data */
+    uint32_t ticket_lifetime_in_secs = 0;
+    RESULT_GUARD(s2n_generate_ticket_lifetime(conn, conn->ticket_fields.key_intro_time, conn->ticket_fields.current_time, &ticket_lifetime_in_secs));
+    /**
+     *= https://www.rfc-editor.org/rfc/rfc8446#section-4.6.1
+     *# The value of zero indicates that the ticket should be discarded immediately.
+     **/
+    RESULT_ENSURE(ticket_lifetime_in_secs > 0, S2N_ERR_SESSION_TICKET_LIFETIME_EXPIRED);
```

Comment:
You marked S2N_ERR_SESSION_TICKET_LIFETIME_EXPIRED as a "usage" error. Usage errors are triggered by the customer / application. Is this error triggered by the application? Can the application control whether this error occurs, and can they take action if they receive it? Maybe more importantly-- CAN an application receive this error?